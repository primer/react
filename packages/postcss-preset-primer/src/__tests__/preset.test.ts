/**
 * @jest-environment node
 */

import postcss from 'postcss'
import preset from '../'

async function process(input: string): Promise<string> {
  try {
    const result = await postcss([preset()]).process(input, {
      from: undefined,
    })
    return result.css.toString()
  } catch (error) {
    // Note: it seems like Jest cannot serialize errors thrown by postcss. As a
    // result, we remove problematic fields before throwing the error
    if (error instanceof postcss.CssSyntaxError) {
      // @ts-expect-error it seems the type for this is incorrect and this field
      // does exist and is causing the circular reference issue
      if (error.postcssNode) {
        // @ts-expect-error it seems the type for this is incorrect and this field
        // does exist and is causing the circular reference issue
        delete error.postcssNode
      }
    }
    throw error
  }
}

describe('postcss-preset-primer', () => {
  test('css nesting', async () => {
    const result = await process(`
      .selector {
        &[data-active] {
          color: var(--fgColor-active);
        }
      }
    `)
    expect(result).toMatchSnapshot()
  })

  describe('mixins', () => {
    test('@mixin focusOutline', async () => {
      const result = await process(`
        .selector {
          @mixin focusOutline;
        }
      `)
      expect(result).toMatchSnapshot()
    })

    test('@mixin focusOutlineOnEmphasis', async () => {
      const result = await process(`
        .selector {
          @mixin focusOutlineOnEmphasis;
        }
      `)
      expect(result).toMatchSnapshot()
    })
  })
})
