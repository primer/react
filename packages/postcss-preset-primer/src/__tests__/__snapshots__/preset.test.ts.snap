// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-preset-primer css nesting 1`] = `".selector[data-active]{color:var(--fgColor-active)}"`;

exports[`postcss-preset-primer mixins @mixin focusOutline 1`] = `".selector{box-shadow:none;outline:2px solid var(--focus-outlineColor,var(--color-accent-fg));outline-offset:-2px}"`;

exports[`postcss-preset-primer mixins @mixin focusOutlineOnEmphasis 1`] = `".selector{box-shadow:inset 0 0 0 3px var(--fgColor-onEmphasis,var(--color-fg-on-emphasis));outline:2px solid var(--focus-outlineColor,var(--color-accent-fg));outline-offset:-2px}"`;
